---
title: "Goodman Project 1"
output: html_document
---

In my project, I work with mouse adenovirus type 1 (MAV-1). Specifically, I'm looking at a cellular kinase and its potential role in antiviral defense. Canonically, protein kinase R (PKR) responds to the presence of double-stranded RNA and it becomes activated to phosphorylate the transcription factor eIF2alpha. When phosphorylated, eIF2alpha halts protein synthesis. GCN2 (general control nonderepressible 2) is another kinase that phosphorylates eIF2alpha; it's activated by amino acid deprivation and UV irradiation. However, recent studies have shown that GCN2 is also activated by Sindbis virus (Berlanga et al., 2006) and HIV-1 (Cosnefroy et al., 2013). I hypothesize that GCN2 is also important during MAV-1 infection.

To test my hypothesis, I have been working with GCN2 mutant mice, termed Atchoum mice. These mice were created during a screen done by the Beutler group. They used ENU mutagenesis to create a library of C57BL/6-background mice, each with single point mutation. They then harvested peritoneal macrophages from each strain of mouse and screened them for susceptibility to human adenovirus and mouse cytomegalovirus. They found that the Atchoum strain of mice showed increased susceptibility to infection by both human adenovirus and mouse cytomegalovirus. When they sequenced the genome of the Atchoum mouse, they found that the single point mutation was in the GCN2 gene. Further investigation determined that the mutation caused a missense mutation and caused incorrect posttranslational splicing that resulted in an inactive kinase.

To test my hypothesis, I started with attempting to recapitulate the macrophage experiments as done by the Beutler group. I harvested peritoneal macrophages from Atchoum and C57BL/6 mice and plated them in 6 well plates. I infected them the next day with MAV-1 at an MOI of 1. After 24, 48, and 72 hours, I harvested individual wells of cells (each is one replicate) and separated the cells from the supernatant. I purified DNA from both the cell and supernatant fraction and used qPCR to quantify the viral loads.

I get the output in an excel document, which I converted to text documents, one for the cell fraction data and one for the supernatant fraction data. The first six lines of each are listed below, cell data first. 

```{r echo=FALSE}
celldata <- read.table(file="qPCR_Raw_Data_Cells.txt", header=T)
superdata <- read.table(file="qPCR_Raw_Data_Supernatant.txt", header=T)
head(celldata)
head(superdata)
```

Let's start with the cell fraction data. I don't get quantifiable levels of DNA from the cell fraction so I have to use housekeeping genes to normalize the amount of DNA over all of the samples. I use GAPDH, so I get GAPDH output in my data. Since I use the qPCR machine program to normalize the data, I no longer need the GAPDH information, I only need the mE1A data (viral load data). So I want to delete all of the rows that contain GAPDH data. The first six rows of the modified data set are shown below.

```{r echo=FALSE}
celldata <- celldata[celldata$target_name == "mE1A",]
head(celldata)
```

